### ISSUE TITLE
Setting Timezone to Berlin sets it to Longyearbyen

### ISSUE THREAD

**Issue by soner89** [saltstack/salt#47559]
### Description of Issue/Question
I want to set the Timezone of a Windows server via Salt to Europe/Berlin. Setting the value returns "True" (=correct behaviour).
Retrieving the Timezone of this server would return "Arctic/Longyearbyen" which is the exact the same time as in Berlin. I expected to retrieve "Europe/Berlin" instead of "Arctic/Longyearbyen".

### Setup
Salt Master: 2016.11.3
Salt Minion: 2016.11.3

### Steps to Reproduce Issue
On the Salt-Master:
```
# salt my_minion timezone.set_zone 'Europe/Berlin'
my_minion:
    True

# salt my_minion timezone.get_zone
my_minion:
    Arctic/Longyearbyen
```

### Versions Report
```
Salt Version:
           Salt: 2016.11.3
 
Dependency Versions:
           cffi: 1.6.0
       cherrypy: Not Installed
       dateutil: 2.6.1
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: 0.24.6
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: 0.24.2
         Python: 2.7.5 (default, Aug  4 2017, 00:39:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.4.1708 Core
        machine: x86_64
        release: 3.10.0-229.11.1.el7.x86_64
         system: Linux
        version: CentOS Linux 7.4.1708 Core

```

**Comment by garethgreenaway** [comment]
@soner89 Thanks for the report.  
@dwoz @twangboy Thoughts on this one?

**Comment by terminalmage** [comment]
This seems to be because `tzutil /g` returns `W. Europe Standard Time`. We're trying to match them to IANA timezones via a dictionary which maps IANA timezones to the Windows timezone. The issue is that `W. Europe Standard Time` maps to _many_ timezones, so the one we return may not be accurate in cases where the same Windows timezone maps to multiple IANA timezones.

@dwoz @twangboy I found [this gist](https://gist.github.com/CMCDragonkai/a53df086cce25319d7530eb4b2ca1da9) which suggests that `tzutil` can give you a UTC offset and country name(s), instead of just a timezone name. This may be a more accurate way of matching the Windows timezone to an IANA timezone.

**Comment by dwoz** [comment]
@terminalmage I agree we need to look at using utc offset and country name or something of that nature. I've assigned this to myself.

**Comment by rallytime** [comment]
@soner89 This should be fixed with #47788. Can you confirm?

**Comment by soner89** [comment]
I just tested the timezones.
![image](https://user-images.githubusercontent.com/3781642/43203728-47eb9bd8-901f-11e8-9924-f9284bd78d7a.png)

Well, it's better than setting it to Longyearbyen but I still would expect the timezone to be set to Berlin and not Vienna, Monaco or Luxembourg.

As I understood, the fix is merged to 2018.3. I installed the latest version (2018.3.2) so the fix should already be implemented, right?
